### PR TITLE
IC-1874: Allow users to go back to EOSR creation page after clicking the "back" button.

### PR DESCRIPTION
**Pull request by Gweaton**
## What does this pull request do?

Ensures "Create End of Service Report" link is displayed if a user clicks the "back" button from that page.

As part of this, we want to ensure that we don't try and create a new End of Service report here when clicking this button if one already exists - we could implement this logic on the backend POST endpoint to return the existing End of Service report, so that it won't try and create a duplicate EOSR, but this felt like an easy approach for now.

## What is the intent behind these changes?

To stop users being stuck and unable to create an end of service report.

### CODE REVIEW

File: server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts [hunk 1]
```diff
@@ -678,12 +678,23 @@ export default class ServiceProviderReferralsController {
...
 
   async createDraftEndOfServiceReport(req: Request, res: Response): Promise<void> {
-    const draftEndOfServiceReport = await this.interventionsService.createDraftEndOfServiceReport(
-      res.locals.user.token.accessToken,
-      req.params.id
-    )
+    const { accessToken } = res.locals.user.token
+    const referralId = req.params.id
+
+    const referral = await this.interventionsService.getSentReferral(accessToken, referralId)
```
Comment (Gweaton) [inline server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts:684]: @tomsmyers do you think it's fine to do this on the Frontend, or should we update the API endpoint to return an existing EOSR if one already exists when we call `createDraftEndOfServiceReport`?

Comment (tomsmyers) [reply server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts:684]: we can do this here, but we should also not allow more than one EOSR to be created for the referral in the API. currently it just creates another.

File: server/routes/serviceProviderReferrals/interventionProgressPresenter.ts [hunk 2]
```diff
@@ -185,11 +185,7 @@ export default class InterventionProgressPresenter {
     ? 'active'
     : 'inactive'
 
-  private get endOfServiceReportSubmitted() {
+  get endOfServiceReportSubmitted(): boolean {
     return this.referral.endOfServiceReport !== null && this.referral.endOfServiceReport.submittedAt !== null
```
Comment (tomsmyers) [inline server/routes/serviceProviderReferrals/interventionProgressPresenter.ts:189]: this can be simplified using `?.`

**Review by tomsmyers — COMMENTED** [review]
i don't see the code here to allow the user to get back to the EOSR creation page?

**Comment by tomsmyers** [comment]
ah, disregard my comment above. i understand now. 

should we follow the pattern we used with action plans here instead? whereby it's created, and once it's created the link no longer says "create", but "continue".

**Comment by tomsmyers** [comment]
> @tomsmyers I've pushed a few changes to this for the time being e.g. "Continue" vs "create" - there's still some logic to work on for when all sessions have been completed etc. but as that's waiting for possible back-end changes, should we get this in as-is, as it's at least an improvement.

yeh sure thing - thanks

File: server/routes/serviceProviderReferrals/interventionProgressPresenter.ts [hunk 3]
```diff
@@ -185,11 +185,23 @@ export default class InterventionProgressPresenter {
     ? 'active'
     : 'inactive'
 
-  private get endOfServiceReportSubmitted() {
-    return this.referral.endOfServiceReport !== null && this.referral.endOfServiceReport.submittedAt !== null
+  get endOfServiceReportSubmitted(): boolean {
+    return !!this.referral.endOfServiceReport?.submittedAt
```
Comment (tomsmyers) [inline server/routes/serviceProviderReferrals/interventionProgressPresenter.ts:189]: not sure i'm a fan of `!!` but i'll roll with it !(!)

**Review by tomsmyers — APPROVED** [review]

**Pull request merged**